spring-session/src/main/java/org/springframework/session/data/redis/RedisOperationsSessionRepository.java

```diff
@@ -540,7 +543,9 @@ private void cleanupPrincipalIndex(RedisSession session) {
 			return;
 		}
 		String sessionId = session.getId();
-		String principal = PRINCIPAL_NAME_RESOLVER.resolvePrincipal(session);
+		Map<String, String> indexes = RedisOperationsSessionRepository.this
+				.indexResolver.resolveIndexesFor(session);
```

Comment:
This would benefit from `IndexResolver` having a method to resolve single index.